src/noggit/scripting/script_model.cpp

```diff
@@ -0,0 +1,248 @@
+// This file is part of Noggit3, licensed under GNU General Public License (version 3).
+#include <noggit/scripting/script_model.hpp>
+#include <noggit/scripting/scripting_tool.hpp>
+#include <noggit/scripting/script_context.hpp>
+#include <noggit/scripting/script_heap.hpp>
+#include <noggit/scripting/script_heap.hpp>
+#include <noggit/scripting/script_exception.hpp>
+
+#include <noggit/World.h>
+#include <noggit/ModelInstance.h>
+#include <noggit/WMOInstance.h>
+#include <noggit/ui/ObjectEditor.h>
+#include <boost/algorithm/string/predicate.hpp>
+
+namespace noggit
+{
+  namespace scripting
+  {
+    static const WMOInstance* wmo_const(script_model const& model)
+    {
+      return (const WMOInstance*)model._model;
+    }
+
+    static const ModelInstance* m2_const(script_model const& model)
+    {
+      return (const ModelInstance*)model._model;
+    }
+
+    static WMOInstance* wmo(script_model& model)
+    {
+      return (WMOInstance*)model._model;
+    }
+
+    static ModelInstance* m2(script_model& model)
+    {
+      return (ModelInstance*)model._model;
+    }
+
+    script_model::script_model(ModelInstance*model)
+      : _model((void*)model), _is_wmo(false) {}
+
+    script_model::script_model(WMOInstance*model)
+      : _model((void*)model), _is_wmo(true) {}
+
+    math::vector_3d model_get_pos(script_model const& model)
+    {
+      if (model._is_wmo)
+      {
+        return wmo_const(model)->pos;
+      }
+      else
+      {
+        return m2_const(model)->pos;
+      }
+    }
+
+    void model_set_pos(script_model& model, math::vector_3d& pos)
+    {
+      if (model._is_wmo)
+      {
+        wmo(model)->pos = pos;
+      }
+      else
+      {
+        m2(model)->pos = pos;
+      }
+    }
+
+    math::vector_3d model_get_rot(script_model const& model)
+    {
+      if (model._is_wmo)
+      {
+        return wmo_const(model)->dir;
+      }
+      else
+      {
+        return m2_const(model)->dir;
+      }
+    }
+
+    void model_set_rot(script_model& model, math::vector_3d& rot)
+    {
+      if (model._is_wmo)
+      {
+        wmo(model)->dir = rot;
+      }
+      else
+      {
+        m2(model)->dir = rot;
+      }
+    }
+
+    float model_get_scale(script_model const& model)
+    {
+      if (model._is_wmo)
+      {
+        return 1;
+      }
+      else
+      {
+        return m2_const(model)->scale;
+      }
+    }
+
+    void model_set_scale(script_model& model, float scale)
+    {
+      if (model._is_wmo)
+      {
+        return;
```

Comment:
throw instead?

Reply:
Really torn on this one. On one hand, a poorly written batch script to randomly resize props that breaks as soon as it hits a wmo, and the user isn't the author so they're stuck. On the other hand, users getting very confused not knowing you can't resize wmo's. Should scripts really be what educates users on this?

My reasoning with the exceptions we do throw is that scripts shouldn't ever end up in the situation where they're reading incorrect script parameters (empty strings etc) and try to use them in the program, while situations like these rely so heavily on where in the game world you're testing that it's a mistake I can see many more people make.

Maybe the best thing is renaming the function `resize_if_m2`

Reply:
I guess a stern warning in the documentation might also be enough. no hard feelings on this one.

Reply:
a74e0df645cb0cc10e065efcfa39725eaf502529